src/client/Microsoft.Identity.Client/ManagedIdentity/AbstractManagedIdentity.cs

```diff
@@ -119,21 +119,88 @@ protected ManagedIdentityResponse GetSuccessfulResponse(HttpResponse response)
             return managedIdentityResponse;
         }
 
-        internal static string GetMessageFromErrorResponse(HttpResponse response)
+        internal string GetMessageFromErrorResponse(HttpResponse response)
         {
-            ManagedIdentityErrorResponse managedIdentityErrorResponse = JsonHelper.TryToDeserializeFromJson<ManagedIdentityErrorResponse>(response?.Body);
-
-            if (managedIdentityErrorResponse == null)
+            if (string.IsNullOrEmpty(response?.Body))
             {
                 return MsalErrorMessage.ManagedIdentityNoResponseReceived;
             }
 
+            try
+            {
+                ManagedIdentityErrorResponse managedIdentityErrorResponse = JsonHelper.DeserializeFromJson<ManagedIdentityErrorResponse>(response?.Body);
+                return ExtractErrorMessageFromManagedIdentityErrorResponse(managedIdentityErrorResponse);
+            }
+            catch
+            {
+                return TryGetMessageFromNestedErrorResponse(response.Body);
+            } 
+        }
+
+        private string ExtractErrorMessageFromManagedIdentityErrorResponse(ManagedIdentityErrorResponse managedIdentityErrorResponse)
+        {
+            StringBuilder stringBuilder = new StringBuilder("[Managed Identity] ");
```

Comment:
consider storing this into a `ManagedIdentityPrefix` string const and checking for, 

```cs
if (stringBuilder.Length == ManagedIdentityPrefix.Length)
{
   return $"{MsalErrorMessage.ManagedIdentityUnexpectedErrorResponse}.";
}

```